#VRML_SIM R2021b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.9999999999999999 9.803670943243452e-17 -1.6014472352266694e-16 1.6008003061004228
  position -0.009248024208911021 3.4559769266545497 -0.07973766293068629
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF GYRO RectangleArena {
  floorSize 2.1445 1.4855
  floorTileSize 0.33 0.33
}
DEF MAZE Group {
  children [
    DEF cornerStone Solid {
      translation -0.7425 0.0325 -0.4125
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(4)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
    }
    DEF cornerStone Solid {
      translation 0.7425 0.0325 -0.4125
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(5)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
    }
    DEF cornerStone Solid {
      translation 0.7425 0.0325 0.4125
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(6)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
    }
    DEF cornerStone Solid {
      translation -0.7425 0.0325 0.4125
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(7)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
    }
    DEF hWall Solid {
      translation 0 0.0325 -0.4125
      children [
        DEF hWall Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 0.3
          }
          geometry Box {
            size 1.46 0.065 0.015
          }
        }
      ]
      name "solid(1)"
      boundingObject USE hWall
      physics Physics {
        density 100
      }
    }
    DEF hWall Solid {
      translation 0 0.0325 0.4125
      children [
        DEF hWall Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 0.3
          }
          geometry Box {
            size 1.46 0.065 0.015
          }
        }
      ]
      boundingObject USE hWall
      physics Physics {
        density 100
      }
    }
    DEF vWall Solid {
      translation -0.7425 0.0325 0
      children [
        DEF vWall Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.3
          }
          geometry Box {
            size 0.015 0.065 0.8
          }
        }
      ]
      name "solid(2)"
      boundingObject USE vWall
      physics Physics {
        density 100
      }
    }
    DEF vWall Solid {
      translation 0.7425 0.0325 0
      children [
        DEF vWall Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.3
          }
          geometry Box {
            size 0.015 0.065 0.8
          }
        }
      ]
      name "solid(3)"
      boundingObject USE vWall
      physics Physics {
        density 100
      }
    }
    DEF Pole MazeBlock {
      translation -0.905 0 -0.5775
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5775 0 -0.2475
      name "maze block(20)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4125 0 -0.2475
      name "maze block(21)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.2475 0 -0.2475
      name "maze block(22)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.0825 0 -0.2475
      name "maze block(23)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.2475 0 -0.2475
      name "maze block(25)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.4125 0 -0.2475
      name "maze block(26)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5775 0 -0.0825
      name "maze block(30)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4125 0 -0.0825
      name "maze block(31)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.2475 0 -0.0825
      name "maze block(32)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.0825 0 -0.0825
      name "maze block(33)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.0825 0 -0.0825
      name "maze block(34)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.2475 0 -0.0825
      name "maze block(35)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.4125 0 -0.0825
      name "maze block(36)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.5775 0 -0.0825
      name "maze block(37)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5775 0 0.0825
      name "maze block(40)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4125 0 0.0825
      name "maze block(41)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.2475 0 0.0825
      name "maze block(42)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.0825 0 0.0825
      name "maze block(43)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.0825 0 0.0825
      name "maze block(44)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.2475 0 0.0825
      name "maze block(45)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.4125 0 0.0825
      name "maze block(46)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5775 0 0.2475
      name "maze block(50)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4125 0 0.2475
      name "maze block(51)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.2475 0 0.2475
      name "maze block(52)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.0825 0 0.2475
      name "maze block(53)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.0825 0 0.2475
      name "maze block(54)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.2475 0 0.2475
      name "maze block(55)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.4125 0 0.2475
      name "maze block(56)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.5775 0 0.2475
      name "maze block(57)"
      length 0.015
    }
    DEF hBlock MazeBlock {
      translation -0.66 0 -0.5775
      name "maze block(10)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.33 0 0.0825
      name "maze block(62)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.165 0 -0.0825
      name "maze block(61)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.165 0 0.0825
      name "maze block(60)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.33 0 -0.2475
      name "maze block(59)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.495 0 -0.0825
      name "maze block(58)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.495 0 0.2475
      name "maze block(39)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.165 0 0.2475
      name "maze block(29)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.905 0 -0.33
      rotation 0 1 0 1.5707996938995747
      name "maze block(78)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.4125 0 0
      rotation 0 1 0 1.5707996938995747
      name "maze block(65)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.2475 0 0.165
      rotation 0 1 0 1.5707996938995747
      name "maze block(64)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.2475 0 0
      rotation 0 1 0 1.5707996938995747
      name "maze block(63)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.2475 0 -0.165
      rotation 0 1 0 1.5707996938995747
      name "maze block(49)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.4125 0 -0.165
      rotation 0 1 0 1.5707996938995747
      name "maze block(11)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.4125 0 0.33
      rotation 0 1 0 1.5707996938995747
      name "maze block(48)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.495 0 -0.2475
      name "maze block(4)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.66 0 -0.0825
      name "maze block(6)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0 0 -0.0825
      name "maze block(8)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0 0 0.0825
      name "maze block(9)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.33 0 0.0825
      name "maze block(12)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.33 0 0.2475
      name "maze block(13)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.0825 0 0.33
      rotation 0 1 0 1.5707996938995747
      name "maze block(18)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.0825 0 -0.33
      rotation 0 1 0 1.5707996938995747
      name "maze block(2)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.0825 0 0
      rotation 0 1 0 1.5707996938995747
      name "maze block(17)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.2475 0 -0.165
      rotation 0 1 0 1.5707996938995747
      name "maze block(15)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.5775 0 0.33
      rotation 0 1 0 1.5707996938995747
      name "maze block(1)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.5775 0 0.165
      rotation 0 1 0 1.5707996938995747
      name "maze block(3)"
      length 0.14
    }
  ]
}
DEF z5162972 E-puck {
  translation -0.66 0 -0.33
  rotation 0 1 0 3.141592653589793
  controller "z5162972_MTRN4110_PhaseA"
  camera_width 320
  camera_height 240
  turretSlot [
    Hinge2Joint {
      device [
        PositionSensor {
        }
      ]
      device2 [
        PositionSensor {
        }
      ]
    }
    DEF DS_BACK DistanceSensor {
      rotation 0 1 0 -1.572
      name "ds_back"
    }
    InertialUnit {
      name "imu"
    }
    DEF DS_LEFT DistanceSensor {
      rotation 0 1 0 3.14
      name "ds_left"
    }
    DEF DS_FRONT DistanceSensor {
      rotation 0 1 0 1.5708
      name "ds_front"
    }
    DEF DS_RIGHT DistanceSensor {
      name "ds_right"
    }
  ]
}
